llvm/test/DebugInfo/CAS/AArch64/multiple_cus.ll

```diff
@@ -0,0 +1,106 @@
+; RUN: llc -cas-friendly-debug-info -O0 --filetype=obj --cas-backend --cas=%t/cas --mccas-casid --mtriple=arm64-apple-darwin %s -o %t/multiple_cus.id 2>&1 | FileCheck %s --allow-empty
+; CHECK-NOT: error in backend: CASBackend output round-trip verification error
+
+; ModuleID = 'llvm-link'
+source_filename = "llvm-link"
+target datalayout = "e-m:o-i64:64-i128:128-n32:64-S128"
+target triple = "arm64-apple-macosx14.0.0"
+
+; Function Attrs: mustprogress noinline nounwind optnone ssp uwtable(sync)
+define noundef i32 @_Z4foo2i(i32 noundef %a) #0 !dbg !11 {
+entry:
+  %a.addr = alloca i32, align 4
+  store i32 %a, ptr %a.addr, align 4
+  call void @llvm.dbg.declare(metadata ptr %a.addr, metadata !17, metadata !DIExpression()), !dbg !18
+  %0 = load i32, ptr %a.addr, align 4, !dbg !19
+  %add = add nsw i32 %0, 5, !dbg !20
+  ret i32 %add, !dbg !21
+}
+
+; Function Attrs: nocallback nofree nosync nounwind speculatable willreturn memory(none)
+declare void @llvm.dbg.declare(metadata, metadata, metadata) #1
+
+; Function Attrs: mustprogress noinline nounwind optnone ssp uwtable(sync)
+define noundef i32 @_Z3fooi(i32 noundef %x) #0 !dbg !22 {
+entry:
+  %x.addr = alloca i32, align 4
+  store i32 %x, ptr %x.addr, align 4
+  call void @llvm.dbg.declare(metadata ptr %x.addr, metadata !23, metadata !DIExpression()), !dbg !24
+  %0 = load i32, ptr %x.addr, align 4, !dbg !25
+  %add = add nsw i32 %0, 2, !dbg !26
+  ret i32 %add, !dbg !27
+}
+
+; Function Attrs: mustprogress noinline nounwind optnone ssp uwtable(sync)
+define noundef float @_Z3barf(float noundef %a) #0 !dbg !28 {
+entry:
+  %a.addr = alloca float, align 4
+  store float %a, ptr %a.addr, align 4
+  call void @llvm.dbg.declare(metadata ptr %a.addr, metadata !33, metadata !DIExpression()), !dbg !34
+  %0 = load float, ptr %a.addr, align 4, !dbg !35
+  %mul = fmul float %0, 5.000000e+00, !dbg !36
+  ret float %mul, !dbg !37
+}
+
+; Function Attrs: mustprogress noinline nounwind optnone ssp uwtable(sync)
+define noundef float @_Z4bar2f(float noundef %x) #0 !dbg !38 {
+entry:
+  %x.addr = alloca float, align 4
+  store float %x, ptr %x.addr, align 4
+  call void @llvm.dbg.declare(metadata ptr %x.addr, metadata !39, metadata !DIExpression()), !dbg !40
+  %0 = load float, ptr %x.addr, align 4, !dbg !41
+  %mul = fmul float %0, 2.000000e+00, !dbg !42
+  ret float %mul, !dbg !43
+}
+
+attributes #0 = { mustprogress noinline nounwind optnone ssp uwtable(sync) "frame-pointer"="non-leaf" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="apple-m1" "target-features"="+aes,+crc,+dotprod,+fp-armv8,+fp16fml,+fullfp16,+lse,+neon,+ras,+rcpc,+rdm,+sha2,+sha3,+v8.1a,+v8.2a,+v8.3a,+v8.4a,+v8.5a,+v8a,+zcm,+zcz" }
```

Comment:
Can you delete all unnecessary attributes? (They tend to change over time and make the test hard to maintain)

Reply:
Apologies on the reduction of the test case, yes I can remove the unnecessary attributes